internal/pkg/cli/deploy/workload.go

```diff
@@ -393,24 +409,46 @@ func (img ContainerImageIdentifier) Tag() string {
 }
 
 func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) error {
+	in := &BuildImageArgs{
+		Name:               d.name,
+		WorkspacePath:      d.workspacePath,
+		Image:              d.image,
+		Mft:                d.mft,
+		Out:                out,
+		CustomTag:          d.image.CustomTag,
+		GitShortCommitTag:  d.image.GitShortCommitTag,
+		BuildFunc:          d.repository.BuildAndPush,
+		Login:              d.repository.Login,
+		CheckDockerEngine:  d.docker.CheckDockerEngineRunning,
+		LabeledTermPrinter: d.labeledTermPrinter,
+	}
+	return buildContainerImages(in)
+}
+
+// BuildContainerImages builds the all the images given the build arguments
+func BuildContainerImages(in *BuildImageArgs) error {
+	return buildContainerImages(in)
+}
+
+func buildContainerImages(in *BuildImageArgs) error {
```

Comment:
I feel like we could just do ⬇️ - I don't see why calling an exported function from the same package would be a problem. Is there any concern that I'm not thinking of?
```suggestion
	return BuildContainerImages(in)
}

// BuildContainerImages builds the all the images given the build arguments
func BuildContainerImages(in *BuildImageArgs) error {
```

Reply:
In addition, I feel like a better name would be `BuildAndPushContainerImages` 👀